### PR TITLE
1. Fix typo or grammar errors 2. Modify some of sentences to deliever…

### PR DESCRIPTION
**Pull request by chokobole**
… their meaning clearly 3. Make some of sentences uniformed such as make verb singular form when describing API 4. Remove trailing spaces

### CODE REVIEW

File: docs/Build-for-ARM-Linux-(RPi2).md [hunk 1]
```diff
@@ -11,23 +11,23 @@ sudo apt-get install gcc-arm-linux-gnueabihf g++-arm-linux-gnueabihf
...
-Copy using your favorite tool or `scp` like below.
+This script gives you `build/arm-linux/release/iotjs/iotjs` or `build/arm-linux/debug/iotjs/iotjs`.
+Copy this binary with your favorite tool or `scp` like below.
 
 ```
 scp build/arm-linux/release/iotjs/iotjs pi@(your RPi2 IP):/home/pi/.
 ```
 
-Open a shell and run with your test program.
+Lastly, Open a shell and run with your test program.
```
Comment (LaszloLango) [inline docs/Build-for-ARM-Linux-(RPi2).md]: wrong case after comma: `Lastly, open...`

File: docs/Build-for-Linux.md [hunk 2]
```diff
@@ -91,9 +91,9 @@ no-snapshot
 nuttx-home= (no default value)
 ```
 
-To give options, please use two dashes '--' before the option name as described in following sections.
+To give options, please use two dashes '--' before the option name as described in the following sections.
 
-Options that may need explanations.
+These are options that you may need explanations.
```
Comment (LaszloLango) [inline docs/Build-for-Linux.md]: The original text is better here

File: docs/Build-for-NuttX-(new).md [hunk 3]
```diff
@@ -1,40 +1,40 @@
...
 
-Follow [Building nuttx for iotjs](https://bitbucket.org/seanshpark/nuttx/wiki/Home) page.
+Follow the [Building nuttx for iotjs](https://bitbucket.org/seanshpark/nuttx/wiki/Home) page.
 
 You should first build NuttX before building IoT.js, it depends on system header files. When NuttX tells you that it needs `libtuv` then come here again and continue with below sections.
 
 #### Build IoT.js for NuttX
 
-Giving these options are needed
+These options are needed
```
Comment (LaszloLango) [inline docs/Build-for-NuttX-(new).md]: "Mandatory arguments"?

File: docs/Build-for-NuttX-(new).md [hunk 4]
```diff
@@ -1,40 +1,40 @@
...
 
-Follow [Building nuttx for iotjs](https://bitbucket.org/seanshpark/nuttx/wiki/Home) page.
+Follow the [Building nuttx for iotjs](https://bitbucket.org/seanshpark/nuttx/wiki/Home) page.
 
 You should first build NuttX before building IoT.js, it depends on system header files. When NuttX tells you that it needs `libtuv` then come here again and continue with below sections.
 
 #### Build IoT.js for NuttX
 
-Giving these options are needed
+These options has to be needed.
```
Comment (chokobole) [inline docs/Build-for-NuttX-(new).md]: @LaszloLango I tried to accept your suggestion, any better idea ?

Comment (LaszloLango) [inline docs/Build-for-NuttX-(new).md]: I'm not sure this sentence is grammatically correct. Previous version was better :)

> These options are needed

**Comment by yichoi** [comment]
LGTM

**Pull request merged**